mgr/ConsDefs.pm

```diff
@@ -1182,13 +1182,18 @@
     } else {
 	print "Could not find xml libs\n" if (! $param::quiet);
     }
- #Vc check SSE support
- my $cmd = "touch temp_gccflags.c; $CXX -E -dM -o - temp_gccflags.c | grep -q SSE";
- my $VcCPPFLAGS = " -DVC_IMPL=SSE";
- if ($STAR_HOST_SYS =~ 'gcc432$' || system($cmd)) {# No SSE
-   $VcCPPFLAGS = " -DVC_IMPL=Scalar";
-   if (-e "temp_gccflags.c") {`rm temp_gccflags.c`;}
- }
+
+    my $VcCPPFLAGS = " -I" . $VC_DIR . "/include" . " -Wabi -fabi-version=0";
+    my $VcLIB = $VC_DIR . "/lib/libVc.a";
+    #Vc check SSE support
+    # my $cmd = "touch temp_gccflags.c; $CXX -E -dM -x c++ -o - temp_gccflags.c | grep -q SSE";
+    my $cmd = "$CXX -E -dM -x c++ - < /dev/null | grep -q SSE";
+    if ($STAR_HOST_SYS =~ 'gcc432$' || system($cmd)) {# No SSE
+        $VcCPPFLAGS .= " -DVC_IMPL=Scalar";
+        # if (-e "temp_gccflags.c") {`rm temp_gccflags.c`;}
+    } else {
+        $VcCPPFLAGS .= " -DVc_IMPL=SSE";
+    }
```

Comment:
You are adding a definition `-DVC_IMPL=...` that is not used in your code.

Reply:
It is not for STAR code, it is used by Vc. This macro is used to control which instruction set should Vc use. Depend on CPU and compiler, If SSE2 and above are available, use SSE, otherwise  fallback to scalar version.

Reply:
Isn't it set automatically if not defined?

Reply:
If not defined, Vc will automatically choose the highest available instruction set. For example, if AVX is available, it will choose AVX. But we only want SSE.

Reply:
Well, in that case what you do here may be inconsistent with how Vc is built by Spack. I think this macro is set to 'auto' by default and that is how Vc lib is built

Reply:
When you build Vc, it builds once for each supported instruction set and archive multiple version of binaries to a signal .a file. When one uses it, we can use the macro Vc_IMPL to pick the desired version.